datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -345,6 +352,10 @@ def session(self):
             if self.tls_use_host_header:
                 self._session.mount('https://', host_header_ssl.HostHeaderSSLAdapter())
 
+            # Enable Unix Domain Socket (UDS) support.
+            # See: https://github.com/msabramo/requests-unixsocket
+            self._session.mount('{}://'.format(UDS_SCHEME), requests_unixsocket.UnixAdapter())
```

Comment:
Shall we check for `is_uds_url` before mounting ? Is it ok to mount for MacOS/Windows?